fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/properties/FDBRecordContextPropertiesTest.java

```diff
@@ -0,0 +1,55 @@
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.properties;
+
+import com.apple.foundationdb.record.RecordCoreException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class FDBRecordContextPropertiesTest {
+    @Test
+    void testDefaultPropertyValue() {
+        final FDBPropertyKey.Builder<Boolean> propertyKeyBuilder = LuceneRecordContextProperties.getLuceneIndexCompressionEnabled();
+        // Build the properties wrapper without configuring the prop with a value supplier
+        final FDBRecordContextProperties properties = FDBRecordContextProperties.newBuilder().build();
+        // Assert the prop value is returned as default value FALSE
+        Assertions.assertEquals(false, properties.getProperty(propertyKeyBuilder), "Unexpected prop value");
+    }
+
+    @Test
+    void testPropertyValueFromSupplier() {
+        final FDBPropertyKey.Builder<Boolean> propertyKeyBuilder = LuceneRecordContextProperties.getLuceneIndexCompressionEnabled();
+        // Build the properties wrapper with configuring the prop with a supplier
+        final FDBRecordContextProperties properties = FDBRecordContextProperties.newBuilder().addProp(propertyKeyBuilder.getName(), propertyKeyBuilder.setValueSupplier(() -> true).build()).build();
+        // Assert the prop value is returned from supplier as TRUE
+        Assertions.assertEquals(true, properties.getProperty(propertyKeyBuilder), "Unexpected prop value");
+    }
+
+    @Test
+    void testInvalidPropType() {
+        final FDBPropertyKey.Builder<Boolean> propertyKeyBuilder = LuceneRecordContextProperties.getLuceneIndexCompressionEnabled();
+        final String propName = propertyKeyBuilder.getName();
+        // Build the properties wrapper with the boolean prop using a String type value
+        final FDBRecordContextProperties properties = FDBRecordContextProperties.newBuilder().addProp(propName, FDBPropertyKey.stringPropertyBuilder(propName, "invalidStringForBooleanProp").build()).build();
+        // Assert exception is thrown due to casting a String to Boolean
+        Assertions.assertThrows(RecordCoreException.class, () -> properties.getProperty(propertyKeyBuilder), "No expected exception");
```

Comment:
I think the message "no expected exception" should probably be either removed or replaced with something like "expect a class cast exception from putting a string in a boolean property".

Reply:
Fixed.